### PR TITLE
First attempt to expose a functor to handle non-manifold elements

### PR DESCRIPTION
**Pull request by nmellado**
_Check if you branch history is PR compatible_
- Your branch need to be up to date with origin/master AND to have linear history (i.e. no merge commit).
- Update your git repository `git fetch origin` if origin is this remote
- Check with the script provided in `scripts/is-history-pr-compatible.sh`
- You must use clang-format style
_These checks are enforced by github workflow actions_
_Please refer to the corresponding log in case of failure_

_UPDATE the form below to describe your PR._

* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
See issue #672 

* **What is the new behavior (if this is a feature change)?**
Expose functor to allo for non-manifold edges processing

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No change on the client size.

* **Other information**:

### CODE REVIEW

File: src/Core/Geometry/TopologicalMesh.cpp [hunk 1]
```diff
@@ -196,197 +160,22 @@ void copyAttribToCore( TriangleMesh& triMesh, const HandleAndValueVector<T>& dat
     }
 }
 
-template <typename T>
-void copyAttribToWedgeData( const TriangleMesh& mesh,
-                            unsigned int vindex,
-                            const std::vector<AttribHandle<T>>& attrHandleVec,
-                            VectorArray<T>* to ) {
-    for ( auto handle : attrHandleVec )
+struct DefaultNonManifoldMeshFunctor
```
[the remark this answers is not in the document]
Comment (dlyr) [reply src/Core/Geometry/TopologicalMesh.cpp:163]:
```suggestion
struct DefaultNonManifoldFaceCommand
```

File: src/Core/Geometry/TopologicalMesh.cpp [hunk 2]
```diff
@@ -196,197 +160,22 @@ void copyAttribToCore( TriangleMesh& triMesh, const HandleAndValueVector<T>& dat
...
 
-template <typename T>
-void copyAttribToWedgeData( const TriangleMesh& mesh,
-                            unsigned int vindex,
-                            const std::vector<AttribHandle<T>>& attrHandleVec,
-                            VectorArray<T>* to ) {
-    for ( auto handle : attrHandleVec )
+struct DefaultNonManifoldMeshFunctor 
+{
+    inline void setTriangleMesh( const TriangleMesh& /*triMesh*/ ) {}
```
[the remark this answers is not in the document]
Comment (dlyr) [reply src/Core/Geometry/TopologicalMesh.cpp:165]:
```suggestion
    inline void process( const TriangleMesh& /*triMesh*/ ) {}
```

**Review by dlyr — CHANGES_REQUESTED** [review]

File: src/Core/Geometry/TopologicalMesh.cpp [hunk 3]
```diff
@@ -196,197 +160,22 @@ void copyAttribToCore( TriangleMesh& triMesh, const HandleAndValueVector<T>& dat
...
+    inline void addNonManifoldFace( const std::vector<TopologicalMesh::VertexHandle>& /*face_vhandles*/)
```
Comment (dlyr) [inline src/Core/Geometry/TopologicalMesh.cpp:166]:
```suggestion
    inline void process( const std::vector<TopologicalMesh::VertexHandle>& /*face_vhandles*/) 
```

**Review by dlyr — APPROVED** [review]

File: src/Core/Geometry/TopologicalMesh.hpp [hunk 4]
```diff
@@ -69,6 +69,19 @@ class RA_CORE_API TopologicalMesh : public OpenMesh::PolyMesh_ArrayKernelT<Topol
...
      * \note This is a costly operation.
+     *
+     * \tparam NonManifoldFaceCommand Command executed when non-manifold faces are
+     * found. API and default implementation:
+     * \snippet Core/Geometry/TopologicalMesh.cpp Default command implementation
+     *
+     */
+    template <typename NonManifoldFaceCommand>
+    explicit TopologicalMesh( const Ra::Core::Geometry::TriangleMesh& triMesh,
+                              NonManifoldFaceCommand functor );
```
Comment (nmellado) [inline src/Core/Geometry/TopologicalMesh.hpp:80]:
```suggestion
                              NonManifoldFaceCommand command );
```

**Review by nmellado — COMMENTED** [review]
Minor remark that might affect the user documentation

**Review by dlyr — APPROVED** [review]

**Review by dlyr — APPROVED** [review]

**Pull request merged**